### PR TITLE
Async warning cleanup

### PR DESCRIPTION
**Pull request by Geobert**
I've fixed most of the clippy's warnings and I'm left with 2 that I don't think we should bother with:
```
warning: you should consider adding a `Default` implementation for `code_grant::resource::Resource`
   --> oxide-auth\src\code_grant\resource.rs:178:5
    |
178 | /     pub fn new() -> Self {
179 | |         Resource {
180 | |             state: ResourceState::New,
181 | |         }
182 | |     }
    | |_____^
    |
    = note: `#[warn(clippy::new_without_default)]` on by default
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#new_without_default
help: try this
    |
176 | impl Default for code_grant::resource::Resource {
177 |     fn default() -> Self {
178 |         Self::new()
179 |     }
180 | }
    |

warning: you seem to be trying to match on a boolean expression
   --> oxide-auth\src\primitives\registrar.rs:316:9
    |
316 | /         match valid {
317 | |             true => Ok(()),
318 | |             false => Err(RegistrarError::Unspecified), 
319 | |         }
    | |_________^ help: consider using an `if`/`else` expression: `if valid { Ok(()) } else { Err(RegistrarError::Unspecified) }`
    |
    = note: `#[warn(clippy::match_bool)]` on by default
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#match_bool   

```

### CODE REVIEW

**Comment by Geobert** [comment]
@HeroicKatora: Here is the clippy tidy up :)

File: oxide-auth/src/code_grant/accesstoken.rs [hunk 1]
```diff
@@ -199,7 +199,7 @@ pub enum Input<'req> {
     /// Positively answer an authentication query.
     Authenticated,
     /// Provide the queried refresh token.
-    Recovered(Option<Grant>),
+    Recovered(Box<Option<Grant>>),
```
Comment (197g) [inline oxide-auth/src/code_grant/accesstoken.rs]: This should probably be an `Option<Box<Grant>>` which has the same size but a much cheaper `None` representation.

**Review by 197g — COMMENTED** [review]
Very small nit, otherwise fine. Those are PRs to your own repo though so you'll need to merge them :) 

Sorry for taking a while to actually review this.

File: oxide-auth/src/code_grant/accesstoken.rs (hunk 1 above)
Comment (Geobert) [reply oxide-auth/src/code_grant/accesstoken.rs]: Done, I've boxed all the Grant for uniformity and push the deref as late as possible.

**Comment by Geobert** [comment]
> Sorry for taking a while to actually review this.

No worries, just came back from holidays today anyway ;) 

My merge plan was:

You merge `async-rebase` in `next`, then I can create a PR to merge `async-sendable` into `next` and then a PR for `async-warning-cleanup` in `next`.

What do you think?

**Comment by 197g** [comment]
Sounds good, I merged the first PR :+1:

**Pull request merged**